### PR TITLE
Travis update and related fixes

### PR DESCRIPTION
**Pull request by amiclaus**
Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
V2:
- add missing brackets

**Review by amiclaus — COMMENTED** [review]
V3:
- update timer_init function, fixing memory leak

File: drivers/platform/aducm3029/timer.c [hunk 1]
```diff
@@ -110,16 +110,18 @@ static void tmr_callback(void *param, uint32_t tmr_event, void *arg)
...
 {
+	struct timer_desc *descriptor;
 	struct aducm_timer_desc *aducm_desc;
 	ADI_TMR_CONFIG		tmr_conf;
 
 	if (!desc || !param || param->freq_hz > FREQ_1MHZ)
 		return FAILURE;
 
-	if (!(*desc = calloc(1, sizeof(**desc))))
+	if (!(descriptor = calloc(1, sizeof(*descriptor)))) {
```
Comment (mchindri) [inline drivers/platform/aducm3029/timer.c:120]: nitpick: I usually use ldesc as (local desc) just to have a short name :)

File: drivers/platform/aducm3029/timer.c [hunk 2]
```diff
@@ -110,16 +110,18 @@ static void tmr_callback(void *param, uint32_t tmr_event, void *arg)
...
 		return FAILURE;
+	}
```
Comment (mchindri) [inline drivers/platform/aducm3029/timer.c:122]: {} are not needed.
Also, not related to your commit but if you can fix my nasty styling: 
```
ldesc =calloc(1, sizeof(*descriptor));
if (!ldesc)
      return FAILURE; 
```

**Review by mchindri — APPROVED** [review]

**Comment by mirceacaprioru** [comment]
Travis is no longer free to use. Maybe check github actions to consider for a shift from Travis CI.

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Review by amiclaus — COMMENTED** [review]
V4:
- remove unnecessary brackets
- rename `descriptor` to `ldesc`

**Pull request merged**